money-core/src/main/scala/com/comcast/money/core/CoreSpanFactory.scala

```diff
@@ -18,12 +18,29 @@ package com.comcast.money.core
 
 import com.comcast.money.api.{ Span, SpanFactory, SpanHandler, SpanId }
 
+import scala.util.{ Success, Failure }
+
 import scala.collection.JavaConversions._
 
 class CoreSpanFactory(handler: SpanHandler) extends SpanFactory {
 
   def newSpan(spanName: String): Span = newSpan(new SpanId(), spanName)
 
+  /**
+   * Continues a trace by creating a child span from the given x-moneytrace header
+   * value or a root span if header is malformed.
+   *
+   * @param childName - the name of the child span to create
+   * @param traceContextHeader - value of x-moneytrace header
+   * @return a child span with trace id and parent id from trace context header or a new root span if the
```

Comment:
this is interesting, is this what we want to do?

I would think if the traceContextHeader is present and malformed, that would be an exception condition.

Perhaps at the minimum we log the error here.